### ISSUE TITLE
Inaccuracy with total_rounds_played

### ISSUE THREAD

**Issue by ChetdeJong** [LaihoE/demoparser#49]
I'm not sure if it's bug with parser or something wrong with demos, but I noticed some inaccuracy with `total_rounds_played`. Sometimes it shows event in wrong round.

Example. Frag at 110006 should be in previous round.

```javascript
const events = parseEvent(
	"./demos/vitality-vs-faze-m3-nuke.dem",
	"player_death",
	["is_warmup_period", "team_name", "team_clan_name", "team_rounds_total"],
	["total_rounds_played"]
);
``` 

(index) | attacker_name | tick | total_rounds_played | user_name
-- | -- | -- | -- | --
6 | 'ropz' | 80026 | 11 | 'flameZ'
7 | 'ropz' | 102804 | 14 | 'ZywOo'
8 | 'ropz' | 109736 | 15 | 'Magisk'
9 | 'ropz' | 109814 | 15 | 'flameZ'
10 | 'ropz' | 109859 | 15 | 'Spinx'
11 | 'ropz' | 109897 | 15 | 'ZywOo'
12 | 'ropz' | 110006 | 16 | 'apEX'
13 | 'ropz' | 114951 | 16 | 'flameZ'
14 | 'ropz' | 115508 | 16 | 'Spinx'

Demo:
https://www.hltv.org/matches/2367247/vitality-vs-faze-iem-sydney-2023

Clip:
https://clips.twitch.tv/SilkyRespectfulMallardMVGame-DK4ADev1iF-YlZgN

**Comment by ChetdeJong** [comment]
Likely because it's last kill of round and it counts it as next round. Again, not sure if it's bug or feature 😄  

Workaround to fix it:

Parse with extra prop `round_win_status`, then check for it and adjust
```javascript
const fixed = events.map((frag) => {
	if (frag.round_win_status != 0) {
		return {
			...frag,
			total_rounds_played: frag.total_rounds_played - 1,
		};
	}
	return frag;
});
```

**Comment by LaihoE** [comment]
Thanks for the detailed example. Was able to reproduce and also fix (but this will be tricky).

The problem is: should game events be parsed before entity updates during a tick. Here the parser updates entities (including the total_rounds_played prop) before the game event is emitted.

Not so long ago I implemented it so that entities are always parsed before game events. But looking at this I might have to revert it. Will likely revert in next release

**Comment by LaihoE** [comment]
This should now be fixed in 0.13.0. The parser now chooses the packet order based on the game event.

**Comment by LaihoE** [comment]
Assume this is fixed, re-open if problems appear
